src/main/java/org/apache/commons/lang3/ArrayUtils.java

```diff
@@ -1868,6 +1868,1013 @@ public static void reverse(final short[] array, final int startIndexInclusive, f
+     * @param offset1 the index of the first element to swap
+     * @param offset1 the index of the second element to swap
+     * @throws ArrayIndexOutOfBoundsException if one of the indices is out of range
+     */
+    public static void swap(final byte[] array, int offset1, int offset2) {
+        if (array == null) {
+            return;
+        }
+        swap(array, offset1, offset2, 1);
+    }
+
+    /**
+     * <p>Swaps two elements in the given array.</p>
+     *
+     * <p>This method does nothing for a {@code null} input array.</p>
+     *
+     * @param array  the array to swap, may be {@code null}
+     * @param offset1 the index of the first element to swap
+     * @param offset1 the index of the second element to swap
+     * @throws ArrayIndexOutOfBoundsException if one of the indices is out of range
+     */
+    public static void swap(final double[] array, int offset1, int offset2) {
+        if (array == null) {
+            return;
+        }
+        swap(array, offset1, offset2, 1);
+    }
+
+    /**
+     * <p>Swaps two elements in the given array.</p>
+     *
+     * <p>This method does nothing for a {@code null} input array.</p>
+     *
+     * @param array  the array to swap, may be {@code null}
+     * @param offset1 the index of the first element to swap
+     * @param offset1 the index of the second element to swap
+     * @throws ArrayIndexOutOfBoundsException if one of the indices is out of range
+     */
+    public static void swap(final float[] array, int offset1, int offset2) {
+        if (array == null) {
+            return;
+        }
+        swap(array, offset1, offset2, 1);
+    }
+
+    /**
+     * <p>Swaps two elements in the given array.</p>
+     *
+     * <p>This method does nothing for a {@code null} input array.</p>
+     *
+     * @param array  the array to swap, may be {@code null}
+     * @param offset1 the index of the first element to swap
+     * @param offset1 the index of the second element to swap
+     * @throws ArrayIndexOutOfBoundsException if one of the indices is out of range
+     */
+    public static void swap(final boolean[] array, int offset1, int offset2) {
+        if (array == null) {
+            return;
+        }
+        swap(array, offset1, offset2, 1);
+    }
+
+    /**
+     * <p>Swaps a series of elements in the given array.</p>
+     * 
+     * <p>This method does nothing for a {@code null} input array.</p>
+     * 
+     * @param array the array to swap, may be {@code null}
+     * @param offset1 the index of the first element in the series to swap
+     * @param offset1 the index of the second element in the series to swap
+     * @param len the number of elements to swap starting with the given indices
+     * @throws ArrayIndexOutOfBoundsException if one of the indices (plus the length of the series to swap) is out of range
+     */
+    public static void swap(final boolean[] array,  int offset1, int offset2, int len) {
```

Comment:
Here an in general I miss a statement to overlapping array ranges. Suppose I call
swap([1,2,3], 0,1,2)
if the swap with the lowest index is executed first the result would be
[2,3,1]
if the swapping starts with the highest offset the result would be
[3,1,2]

Should two calls with exchanged offsets should yield the same result?
swap([1,2,3], 0,1,2)  == swap([1,2,3],1, 0, 2)

IMHO yes. This implementation does not guarantee this.